### PR TITLE
Bump version to 1.0.0

### PR DESCRIPTION
**Pull request by robertschaedler3**
## Description

Bump version from `0.4.1` to `0.5.0`.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -5,9 +5,9 @@ cmake_minimum_required(VERSION 3.2)
 
 project(osconfig)
 
-set(MAJOR_VERSION "0" CACHE STRING "Major version to use")
-set(MINOR_VERSION "4" CACHE STRING "Minor version to use")
-set(PATCH_VERSION "1" CACHE STRING "Patch version to use")
+set(MAJOR_VERSION "1" CACHE STRING "Major version to use")
```
Comment (MariusNi) [inline src/CMakeLists.txt:8]: >"Major version to use") [](http://example.com/codeflow?start=35&length=23)

Can we remove "to use"? (which is obvious)

And have just "Major/Minor/Patch version"?

Comment (robertschaedler3) [reply src/CMakeLists.txt:8]: I agree 👍🏻

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**